allensdk/brain_observatory/ecephys/ecephys_session.py

```diff
@@ -506,11 +506,33 @@ def _build_units_table(self, units_table):
             & (table['quality'] == 'good')
         ]
 
-        table = table.drop(columns=['local_index_unit', 'quality', 'valid_data'])
+        # table = table.drop(columns=['local_index_unit', 'quality', 'valid_data'])
         return table.sort_values(by=['probe_description', 'probe_vertical_position', 'probe_horizontal_position'])
 
 
+    def _build_nwb1_waveforms(self, mean_waveforms):
+        # _build_mean_waveforms() assumes every unit has the same number of waveforms and that a unit-waveform exists
+        # for all channels. This is not true for NWB 1 files where each unit has ONE waveform on ONE channel
+        units_df = self.units
+        output_waveforms = {}
+        for uid in list(mean_waveforms.keys()):
+            data = mean_waveforms.pop(uid)
+            output_waveforms[uid] = xr.DataArray(
+                data=data,
+                dims=['channel_id', 'time'],
+                coords={
+                    'channel_id': [units_df.loc[uid]['peak_channel_id']],
+                    'time': np.arange(data.shape[1]) / 30000  # Don't know where this came from
```

Comment:
that's the sampling rate (probe-local clock)! want to make it a parameter / figure out where the value is stored and get it from there?

Reply:
I can go ahead and try to implement sampling rate, but are you sure it exists and can be fetched from the nwb file?

Reply:
Apparently code to calculate the sampling rate will have (albeit slightly) different values for different probes, making it hard to uniformly label an xarray across different channels. So I created two EcephysSessionApi methods for getting the sampling rate - one for the baseline/expected and other for the actual (which we will need to get from xiaoxuan).